src/index.js

```diff
@@ -9,13 +10,25 @@ var shisha = {
      * @param data
      * @param locals
      * @param callback
+     * @param capath
      */
-    smoke: function (data, locals, callback) {
+    smoke: function (data, locals, callback, capath) {
         var report = [];
-        
-        if (!callback) {
-            callback = locals;
-            locals = {};
+
+        if (!capath)
```

Comment:
@mohdhallal could you please explain the logic between line 18 and 31? Are we shifting arguments?